ghengin-core/ghengin-vulkan/Ghengin/Vulkan/Renderer/Buffer.hs

```diff
@@ -101,7 +101,7 @@ destroyDeviceLocalBuffer (DeviceLocalBuffer b dm) = enterD "destroyDeviceLocalBu
 --
 -- This is unlike DeviceLocalBuffers, which are allocated on the device and
 -- require a staging buffer and a copy command to be written
-data MappedBuffer = UniformBuffer { buffer  :: {-# UNPACK #-} !Vk.Buffer
+data MappedBuffer = MappedBuffer { buffer  :: {-# UNPACK #-} !Vk.Buffer
```

Comment:
Is this right? I thought of making it a sum type, with the constructors `UniformBuffer` and `StorageBuffer`, but since they are so similar, they can use the same data. This might not work if PushConstants are to be implemented.

Reply:
Yes, that seems fine since the Uniform vs Storage detail is more relevant indeed as part of each property. MappedBuffer can be more generic as you've done. 👍